src/sphobjinv/cmdline.py

```diff
@@ -774,53 +741,55 @@ def do_convert(inv, in_path, params):
         |dict| -- Parameters/values mapping from the active subparser
 
     """
-    # TODO: Refactor to-file output to new function
-    # TODO: Add tests for stdout output functionality
-    if params[OUTFILE] == "-" or (params[INFILE] == "-" and params[OUTFILE] is None):
-        write_stdout(inv, params)
-        return
-
     mode = params[MODE]
 
     # Work up the output location
-    try:
-        out_path = resolve_outpath(params[OUTFILE], in_path, params)
-    except Exception as e:  # pragma: no cover
-        # This may not actually be reachable except in exceptional situations
-        selective_print("\nError while constructing output file path:", params)
-        selective_print(err_format(e), params)
-        sys.exit(1)
+    if params[OUTFILE] == "-" or (params[INFILE] == "-" and params[OUTFILE] is None):
+        file = sys.stdout.buffer
+        out_path = None
+    else:
+        try:
+            out_path = resolve_outpath(params[OUTFILE], in_path, params)
+        except Exception as e:  # pragma: no cover
+            # This may not actually be reachable except in exceptional situations
+            selective_print("\nError while constructing output file path:", params)
+            selective_print(err_format(e), params)
+            sys.exit(1)
+        file = open(out_path, 'wb')
 
     # If exists, confirm overwrite; clobber if QUIET
-    if os.path.isfile(out_path) and not params[QUIET] and not params[OVERWRITE]:
+    if out_path and os.path.isfile(out_path) and not params[QUIET] and not params[OVERWRITE]:
         resp = yesno_prompt("File exists. Overwrite (Y/N)? ")
         if resp.lower() == "n":
             print("\nExiting...")
             sys.exit(0)
 
-    # Write the output file
+    # Write the output
     try:
         if mode == ZLIB:
-            write_zlib(inv, out_path, expand=params[EXPAND], contract=params[CONTRACT])
+            write_zlib(inv, file, expand=params[EXPAND], contract=params[CONTRACT])
         if mode == PLAIN:
-            write_plaintext(
-                inv, out_path, expand=params[EXPAND], contract=params[CONTRACT]
-            )
+            write_plaintext(inv, file, expand=params[EXPAND], contract=params[CONTRACT])
         if mode == JSON:
-            write_json(inv, out_path, expand=params[EXPAND], contract=params[CONTRACT])
+            write_json(inv, file, expand=params[EXPAND], contract=params[CONTRACT])
+    except BrokenPipeError:
+        pass
     except Exception as e:
         selective_print("\nError during write of output file:", params)
         selective_print(err_format(e), params)
         sys.exit(1)
+    finally:
+        file.close()
```

Comment:
I guess calling `.close()` on `sys.stdout.buffer` doesn't cause a problem for the next executable, trying to read the piped data?

Reply:
No, try it! You can add any pipe here:

```console
$ python -c 'import; b = sys.stdout.buffer; b.write(b"1"); b.close()'
```